Fixed issues #83 and #85

Prayag-X opened the pull request:
Fixed issues #83 and #85 by adding new parameter `allowDragOnlyFromRevealedArea` to `LiquidSwipe` class which is by default false, and so doesn't change any behavior of the package's present version unless the user chooses to enable it. Basically what is changed is that when the parameter is set to true, the drag will only work from the revealed page area and not from the whole page, which was overwriting any horizontal gestures given to the pages within the `LiquidSwipe`. So users, if faces any kind of problems with giving gestures inside the pages can now enable it to fix the issue.

iamSahdeep commented:
Hi @Prayag-X Thanks for the PR, really appreciate it. 
It seems like even after enabling the `allowDragOnlyFromRevealedArea`, I am experiencing the same behaviour  as previously this is, that I could drag the page from anywhere on the whole page.

Can you please update the example so that I can confirm if I am not missing anything. Thanks Again.

Prayag-X commented:
Hello @iamSahdeep

It seems `allowDragOnlyFromRevealedArea` works only when there is any kind of drag inside the page container. But yes, the whole page is draggable when there is no draggable widget inside the page container. 

At first without `allowDragOnlyFromRevealedArea`, any horizontal drag or gesture inside the page container didn't work at all. Though it solves the main issue, it creates another issue. I couldn't figure out a way to make the thing work regardless there is any drag in the page or not. The problem maybe that the whole page becomes the `Gesture arena` when there is no draggable child in the page. Changing any gesture behavior didn't work either. 

Check out the `Troubleshooting` section of the Flutter official doc https://api.flutter.dev/flutter/widgets/GestureDetector-class.html and this doc too https://docs.flutter.dev/development/ui/advanced/gestures#gesture-disambiguation

Try this simple example I made for the testing purpose https://github.com/Prayag-X/liquid_swipe_test

Should I change the parameter name from `allowDragOnlyFromRevealedArea` to `preferDragFromRevealedArea` and modify the comment? I guess that will solve the issues as well as won't create any confusion. I will also update the example with it and create a PR after that.

Prayag-X commented:
@iamSahdeep done

1. Renamed the parameter `allowDragOnlyFromRevealedArea` to `preferDragFromRevealedArea` and updated its documentation.
2. Updated and added another screen for `preferDragFromRevealedArea` to the example.

axijethwa74 reviewed (commented):
what is this bro

Prayag-X commented:
@axijethwa74 

Changes in `liquid_swipe.dart` are for:
1. Addding the new parameter `preferDragFromRevealedArea`. Its purpose is mentioned in the above comments.
2. For its purpose, had to refractor some functions and parameters but in general it doesn't change any normal functionality.

Changes in `PageHelpers/page_dragger.dart` are for:
1. Adding the gesture detector to the arena of revealed part of the page as a whole instead of only to the sliding icon for the purpose of the new parameter `preferDragFromRevealedArea`.

Changes in `/example` are for:
1. Updated to null safety and so the required changes in files other than `lib/main.dart`
2. Upgraded deprecated widgets like `FlatButton` to `TextButton`
3. Few very minor improvements to the UI.
4. Added a new page for testing `preferDragFromRevealedArea` as suggested by @iamSahdeep 

All the individual changes can also be explained in details

iamSahdeep commented:
Thanks @Prayag-X Awesome work.
Will soon update the pub as well.

Prayag-X commented:
The honor is mine that you accepted my pr @iamSahdeep ❤️
